src/joystick/hidapi/SDL_hidapi_sinput.c

```diff
@@ -0,0 +1,745 @@
+        if (ctx->accelerometer_supported) {
+
+            accel = (Sint16)(data[SINPUT_REPORT_IDX_IMU_ACCEL_Y] | (data[SINPUT_REPORT_IDX_IMU_ACCEL_Y + 1] << 8));
+            imu_values[2] = -(float)accel * ctx->accelScale; // Y-axis acceleration
+
+            accel = (Sint16)(data[SINPUT_REPORT_IDX_IMU_ACCEL_Z] | (data[SINPUT_REPORT_IDX_IMU_ACCEL_Z + 1] << 8));
+            imu_values[1] = (float)accel * ctx->accelScale; // Z-axis acceleration
+
+            accel = (Sint16)(data[SINPUT_REPORT_IDX_IMU_ACCEL_X] | (data[SINPUT_REPORT_IDX_IMU_ACCEL_X + 1] << 8));
+            imu_values[0] = -(float)accel * ctx->accelScale; // X-axis acceleration
+
+            SDL_SendJoystickSensor(timestamp, joystick, SDL_SENSOR_ACCEL, ctx->imu_timestamp, imu_values, 3);
+        }
+
+        // Process Gyroscope
+        if (ctx->gyroscope_supported) {
+            
+            gyro = (Sint16)(data[SINPUT_REPORT_IDX_IMU_GYRO_Y] | (data[SINPUT_REPORT_IDX_IMU_GYRO_Y + 1] << 8));
+            imu_values[2] = -(float)gyro * ctx->gyroScale; // Y-axis rotation
+
+            gyro = (Sint16)(data[SINPUT_REPORT_IDX_IMU_GYRO_Z] | (data[SINPUT_REPORT_IDX_IMU_GYRO_Z + 1] << 8));
+            imu_values[1] = (float)gyro * ctx->gyroScale; // Z-axis rotation
+
+            gyro = (Sint16)(data[SINPUT_REPORT_IDX_IMU_GYRO_X] | (data[SINPUT_REPORT_IDX_IMU_GYRO_X + 1] << 8));
+            imu_values[0] = -(float)gyro * ctx->gyroScale; // X-axis rotation
+
+            SDL_SendJoystickSensor(timestamp, joystick, SDL_SENSOR_GYRO, ctx->imu_timestamp, imu_values, 3);
+        }
+    }    
+
+    SDL_memcpy(ctx->last_state, data, SDL_min(size, sizeof(ctx->last_state)));
+}
+
+static bool HIDAPI_DriverSInput_UpdateDevice(SDL_HIDAPI_Device *device)
+{
+    SDL_DriverSInput_Context *ctx = (SDL_DriverSInput_Context *)device->context;
+    SDL_Joystick *joystick = NULL;
+    Uint8 data[USB_PACKET_LENGTH];
+    int size = 0;
+
+    if (device->num_joysticks > 0) {
+        joystick = SDL_GetJoystickFromID(device->joysticks[0]);
+    } else {
+        return false;
+    }
+
+    while ((size = SDL_hid_read_timeout(device->dev, data, sizeof(data), 0)) > 0) {
+#ifdef DEBUG_SINPUT_PROTOCOL
+        HIDAPI_DumpPacket("SInput packet: size = %d", data, size);
+#endif
+        if (!joystick) {
+            continue;
+        }
+
+        if (!ctx->feature_flags_obtained && !ctx->feature_flags_sent && ctx->player_idx > 0)
+        {
+            // Send feature get command
+            // Set player number, finalizing the setup
+            Uint8 featuresGetCommand[SINPUT_DEVICE_REPORT_COMMAND_SIZE] = { SINPUT_DEVICE_REPORT_ID_OUTPUT_CMDDAT, SINPUT_DEVICE_COMMAND_FEATURES };
+            int featureNumBytesWritten = SDL_HIDAPI_SendRumble(device, featuresGetCommand, SINPUT_DEVICE_REPORT_COMMAND_SIZE);
+
+            if (featureNumBytesWritten < 0) {
+                SDL_SetError("SInput device features get command could not write");
+            }
+            else
+            {
```

Comment:
Can you fix the style in the rest of the file as well?
```suggestion
            } else {
```